### PR TITLE
Initialize HTML elements with array of attributes

### PR DESCRIPTION
**Pull request by lovetodream**
This allows to create custom HTML components and forward attributes from those to the underlying tag.

Example:
```swift
struct Title<Title: HTML>: HTML {
    var attributes: [HTMLAttribute<HTMLTag.h2>]
    var title: () -> Title

    init(attributes: HTMLAttribute<HTMLTag.h2>..., @HTMLBuilder title: @escaping () -> Title) {
        self.attributes = attributes + [.class("mt-14 text-base/7 font-semibold text-zinc-950 sm:text-sm/6 dark:text-white")]
        self.title = title
    }

    var content: some HTML {
        h2(attributes) {
            title()
        }
    }
}
```

Let me know if I should add tests for this.

### CODE REVIEW

**Comment by lovetodream** [comment]
Example CI is failing due to changes in the Hummingbird release candidate

File: Sources/Elementary/Core/Html+Attributes.swift [hunk 1]
```diff
@@ -79,6 +79,15 @@ public extension HTMLElement {
...
     }
+
+    /// Creates a new HTML element with the specified attributes and content.
+    /// - Parameters:
+    ///  - attributes: The attributes to apply to the element as an array.
+    ///  - content: The content of the element.
+    init(_ attributes: [HTMLAttribute<Tag>], @HTMLBuilder content: () -> Content) {
+        self.attributes = .init(attributes)
+        self.content = content()
+    }
```
Comment (sliemeobn) [inline Sources/Elementary/Core/Html+Attributes.swift:90]: I was thinking we should maybe have the label "attributes" here, to better distinguish it from the "normal" variadic version. wdyt?

File: Sources/Elementary/Core/Html+Attributes.swift [hunk 2]
```diff
@@ -117,6 +126,15 @@ public extension HTML where Tag: HTMLTrait.Attributes.Global {
...
     }
+
+    /// Adds the specified attributes to the element.
+    /// - Parameters:
+    ///   - attributes: The attributes to add to the element as an array.
+    ///   - condition: If set to false, the attributes will not be added.
+    /// - Returns: A new element with the specified attributes added.
+    func attributes(_ attributes: [HTMLAttribute<Tag>], when condition: Bool = true) -> _AttributedElement<Self> {
+        _AttributedElement(content: self, attributes: .init(condition ? attributes : []))
+    }
```
Comment (sliemeobn) [inline Sources/Elementary/Core/Html+Attributes.swift:137]: also here, I was thinking a label might be better... maybe `attributes(elemenetsOf: myArray)`?

File: Sources/Elementary/Core/Html+Attributes.swift [hunk 3]
```diff
@@ -79,6 +79,15 @@ public extension HTMLElement {
...
+    /// - Parameters:
+    ///  - attributes: The attributes to apply to the element as an array.
+    ///  - content: The content of the element.
+    init(_ attributes: [HTMLAttribute<Tag>], @HTMLBuilder content: () -> Content) {
+        self.attributes = .init(attributes)
+        self.content = content()
+    }
 }
 
 public extension HTMLVoidElement {
```
Comment (sliemeobn) [inline Sources/Elementary/Core/Html+Attributes.swift:93]: we should add the same new array-based initializer overload to `HTMLVoidElement`

**Review by sliemeobn — CHANGES_REQUESTED** [review]
hey, thanks for this, I was going to add the same thing today ; )

please add a test or two just to have it covered - nothing fancy

also, I am in the process of fixing the hummingbird example right now.

File: Sources/Elementary/Core/Html+Attributes.swift (hunk 1 above)
Comment (lovetodream) [reply Sources/Elementary/Core/Html+Attributes.swift:90]: Sounds good

File: Sources/Elementary/Core/Html+Attributes.swift (hunk 2 above)
Comment (lovetodream) [reply Sources/Elementary/Core/Html+Attributes.swift:137]: what about `attributes(contentsOf: myArray)`, this would mirror other collections' methods?

Comment (sliemeobn) [reply Sources/Elementary/Core/Html+Attributes.swift:137]: oh, yes, that was what I was going for (just remembered it wrong) ; )

**Review by sliemeobn — APPROVED** [review]
nice one, thanks!

lgtm

**Pull request merged**